### ISSUE TITLE
Develop Component resource for: kubernetes:apiextensions.k8s.io:CustomResource

### ISSUE THREAD

**Issue by mjeffryes** [pulumi/pulumi-kubernetes#2787]
`CustomResource` is a generic resource for creating a Kubernetes custom resource; see [example](https://www.pulumi.com/ai/answers/giesKpRK2snfzBpgxfgzw5/creating-kubernetes-custom-resources-with-third-party-apis). Today it is implemented as an 'overlay' resource and has two major limitations:
- not supported in Pulumi YAML; see https://github.com/pulumi/pulumi-yaml/pull/158
- not supported in Pulumi Java SDK

Conceptually,`CustomResource` represents a single Kubernetes object akin to `ConfigMap` or `Deployment`. It's not a component resource and should continue to benefit from the usual resource lifecycle (e.g. diff).

For most use-cases, a reasonable alternative to `CustomResource` is `ConfigGroup` with its `objs` input property, which allows for literal Kubernetes objects.  That said, a component resource is an awkward substitute for a custom resource. 

Note that the Pulumi Converter for Kubernetes (kube2pulumi and pulumi-converter-kubernetes) doesn't support custom resources (see [Limitations](https://github.com/pulumi/pulumi-converter-kubernetes/tree/main?tab=readme-ov-file#limitations)).
```[tasklist]
### Resolves
- [ ] https://github.com/pulumi/pulumi-kubernetes/issues/1082
- [ ] https://github.com/pulumi/pulumi-kubernetes/issues/1115
```

**Comment by EronWright** [comment]
I was curious as to why Pulumi YAML is unable to simply use the `CustomResource` as provided by the kubernetes provider without an overlay. Imagine we removed the special-case logic ([here](https://github.com/pulumi/pulumi-yaml/blob/v1.6.0/pkg/pulumiyaml/packages.go#L232)) and allowed the YAML engine to register the resource. We see the essential problem is a limitation of Pulumi schema: lack of support for open-ended types w/ extra properties.

For example, given a YAML program:
```yaml
resources:
  myCustomResource:
    type: kubernetes:apiextensions.k8s.io:CustomResource
    properties:
      apiVersion: stable.example.com/v1
      kind: CronTab
      metadata:
        name: my-new-cron-object
      spec:
        cronSpec: "* * * * */5"
```

The YAML engine obtains schema info from the provider, and is unable to handle the `spec`  property.  
```
    Error: Property spec does not exist on 'kubernetes:apiextensions.k8s.io:CustomResource'
      on Pulumi.yaml line 16:
      16:       spec:
    Cannot assign '{apiVersion: string, kind: string, metadata: {name: string}, spec: {cronSpec: string, image: string}}' to 'kubernetes:apiextensions.k8s.io:CustomResource':
      Existing properties are: kind, others, apiVersion, metadata
```

Observe that the schema for `CustomResource` ([overlays.go](https://github.com/pulumi/pulumi-kubernetes/blob/0b393b10d06fe630d4ab8a719e10e31958c13a29/provider/pkg/gen/overlays.go#L1179-L1207)) has the standard `kind`, `apiVersion`, and `metadata` properties.  Then there's a `others` property (a map type), which is how the magic happens. The overlay code packs the `spec` into `others` to convey it to the provider. [Here's](https://github.com/pulumi/pulumi-kubernetes/blob/v4.9.0/sdk/nodejs/apiextensions/customResource.ts#L124) how the overlay type takes the input.

In other words, the SDKs simulate support for open-ended schemas.

**Comment by EronWright** [comment]
We chose to address the use-case behind this issue in a limited way, by implementing `CustomResource` for the Java SDK. 

For the Pulumi YAML SDK, it isn't practical to provide an implementation at this time. Please consider using the `ConfigGroup` resource, i.e. the `objs` property that lets you define Kubernetes objects similarly to `CustomResource`.
